### ISSUE TITLE
Authentication doesn't seem to work

### ISSUE THREAD

**Issue by julianwachholz** [prikhi/lightdm-mini-greeter#2]
Hi! I've finally updated the package and tried using it again, I'm getting up to the login screen with my config all setup with my username, but upon entering my password and pressing enter, nothing happens. And I'm 99.9999% sure it's the right password. :) I couldn't find anything interesting in the logs either.

Also: How do I change the password input from RTL to LTR text?

**Comment by prikhi** [comment]
What OS/distro are you using? And what version of lightdm do you have?

Will have to look into how to switch text input alignment, right now I don't think there's anything explicitly setting it to RTL.

Thanks for your patience & sorry I didn't get to this sooner, wish I could get github to email me when new issues are created >.<

**Comment by prikhi** [comment]
@julianwachholz Polite bump.
Would like to resolve this, can you tell me what distro you use?

**Comment by julianwachholz** [comment]
Sorry I totally forgot. I'm on ArchLinux too. Will update later today!

**Comment by prikhi** [comment]
Huh that's strange, authentication has been working fine for me :/

What version of lightdm are you running(`pacman -Qs lightdm`)? 

Any difference installing the [AUR package](https://aur.archlinux.org/packages/lightdm-mini-greeter/) instead? Dunno how that'd change anything, but worth a try I guess?

If you're system is up to date & it's still not working, would you mind sending me a list of your installed packages(`pacman -Qe`)? If you don't want to post it publicly, feel free to email it to me(click the maintainer link on AUR page to see my email).

Just trying to reproduce this :/ Might just have to add more debugging output to the logs.

**Comment by davidmh** [comment]
Having the same issue as @julianwachholz 

Also on Arch, I'm running the AUR package and this is the output for `pacman -Qs lightdm`

```bash
local/lightdm 1:1.22.0-1
    A lightweight display manager
local/lightdm-mini-greeter 0.1.3-1
    A Minimal, Configurable, Single-User GTK3 LightDM Greeter
```

**Comment by davidmh** [comment]
Managed to find the problem by tailing `/var/log/lightdm/lightdm.log` from tty2 and tried to authenticate again.

```log
[+44.01s] DEBUG: Session pid=919: Authentication complete with return value 0: Success
[+44.01s] DEBUG: Authenticate result for user chaac: Success
[+44.01s] DEBUG: User chaac authorized
[+44.01s] DEBUG: Greeter requests session default
[+44.01s] DEBUG: Seat seat0: Failed to find session configuration default
[+44.01s] DEBUG: Seat seat0: Can't find session 'default'
[+48.13s] DEBUG: Seat seat0 changes active session to c2
```
So looks like it's not an authentication problem, but the inability to find the session.

I had `user-session` defined under the group `[LightDM]`, so I tried adding it under `[Seat:*]` as well, restarted lightdm and it worked!

Hope this helps.

**Comment by prikhi** [comment]
Awesome! I think the proper place is under the `Seat:*` group, you shouldn't need it in `LightDM` as well.

I can add that to the README & text printed out when installing the AUR package, as well as showing an error if the greeter can't find a default session.

**Comment by T-ii** [comment]
Hi, I'm having the same issues. Have it config'd with my username and I'm 100% sure I'm typing the correct password, but it won't log me in. 

Sorry, I'm pretty new to linux so could you describe what you did to fix it? @prikhi @davidmh 

Thank you!

**Comment by davidmh** [comment]
Hi @T-ii, for sure.

1. Once you're in the lightdm prompt screen, open a TTY, you can do that by using `ctr+alt+f2`.

2. Then do a fresh tail to follow the messages in the lightdm log:

```bash
sudo tail -fn0 /var/log/lightdm/lightdm.log
```

3. Switch back to the lightdm promp twith `ctrl+alt+f7` and try to login again.

4. Return to your tail with `ctrl+alt+f2` and see what messages were thrown into the log.

The problem should be described there.

**Comment by T-ii** [comment]
Seems I get similar results as you, @davidmh 

[+65.63s] DEBUG: Session pid=637: Authentication complete with return value 0: Success
[+65.64s] DEBUG: Authenticate result for user nn: Success
[+65.64s] DEBUG: User nn authorized
[+65.64s] DEBUG: Greeter requests session default
[+65.68s] DEBUG: Writing /home/nn/.dmrc
[+65.69s] DEBUG: Seat seat0: Failed to find session configuration default
[+65.69s] DEBUG: Seat seat0: Can't find session 'default'
[+69.67s] DEBUG: Seat seat0 changes active session to c2

Not sure how to proceed with this info. Thank you for the help.

**Comment by davidmh** [comment]
```bash
[+65.69s] DEBUG: Seat seat0: Can't find session 'default'
```
It means you need to specify what xsession you want to start after a successful login.

Look for the `user-session` example under `[Seat:*]` and set it to whatever xsession you want.

From [this Ubuntu wiki](https://wiki.ubuntu.com/LightDM#Changing_the_Default_Session):

> user-session=name
Where **name** is the name of the session .desktop file from /usr/share/xsessions/*.desktop.

**Comment by T-ii** [comment]
Wow I'm a complete idiot. I'll leave this up here for anyone else having the problem...

I found that solution and tried many times to change my 'user-session', but because every single file in /usr/share/xsessions was ended with a .desktop extension, that's what I put.... 
`user-session=gnome.desktop`

I just had to remove the .desktop extension.... Doh. 

Thank you for all the help! LightDM mini greeter is fantastic so far!

**Comment by necauqua** [comment]
Had same issue, fixed it myself, fun to accidentally find it here))
I thought it would error somehow myself, so waiting for an update :)

**Comment by prikhi** [comment]
Been away, PKGBUILD now prints what options you must change, will add error soon

**Comment by prikhi** [comment]
Should be fixed in a28ff60ace80b9e8804bfbf370393ded290e8d02

Instead of showing an error, I fixed launching the default user session. So if you only have 1 DE/WM installed, you shouldn't need to set the `user-session` option in `lightdm.conf` anymore.

Lemme know if this is inadequate in some way. Gonna test & merge the font-size PR &  then I'll release a new version.
